packages/core/src/i18n.test.ts

```diff
@@ -198,15 +197,22 @@ describe("I18n", () => {
     })
 
     expect(i18n._("Hello")).toEqual("Salut")
-    expect(i18n._("My name is {name}", { name: "Fred" })).toEqual(
-      "Je m'appelle Fred"
-    )
+    expect(
+      i18n._({
+        id: "My name is {name}",
+        message: "Je m'appelle {name}",
+        values: { name: "Fred" },
+      })
+    ).toEqual("Je m'appelle Fred")
```

Comment:
That's what i'm mentioned. Despite the fact that catalog should always be compiled, there might be users which may use not compiled catalog in development, because it just works now. I'm a little concerned about merging it into a `main` branch. But we can merge it to the `next` and make a beta release with this. This will be planned for v5 and documented in the migration guide.

Reply:
I completely agree with your perspective. Judging from the way other unit tests in the current repository are written, it seems like this 'feature' could be easily adopted. It's safer to proceed with the approach in the `next` branch.